### PR TITLE
feature/476

### LINKED ISSUE #476: Add option in settings to enable loading of 'wip' airports
At the end of this sprint, any airports not in compliance with the new specs will be removed from the airport selection dialog. So as not to make regular users furious, an option should be added in the settings menu that allows them to access those airports after enabling "Show WIP airports".

For those reading, please note that this will be a temporary measure. Once we get a decent number of airports back into the main load list, this option (and the ability to access those airports) will be removed.

---

- [x] add option in settings to show/not show WIP airports
- [x] conditionally render airport list based on that game option
- [x] verify update when option changes

### PR DESCRIPTION
**Pull request by n8rzz**
Resolves #476.

Add option in settings to enable loading of 'wip' airports

### CODE REVIEW

File: src/assets/scripts/client/UiController.js [hunk 1]
```diff
@@ -293,95 +323,123 @@ export default class UiController {
...
-                    break;
-                case 'expert':
-                    difficulty = icon.repeat(5);
-                    break;
-                default:
-                    difficulty = '?';
-                    break;
+            const { name, icao, level, wip } = this._airportController.airport.airports[airports[i]];
+
+            if (!shouldShowWipAirports) {
```
Comment (n8rzz) [inline src/assets/scripts/client/UiController.js]: :facepalm:

this logic is wrong

File: src/assets/scripts/client/constants/eventNames.js [hunk 2]
```diff
@@ -21,5 +21,11 @@ export const EVENT = {
...
      * @type {string}
      */
-    SHOULD_PAUSE_UPDATE_LOOP: 'should-pause-update-loop'
+    SHOULD_PAUSE_UPDATE_LOOP: 'should-pause-update-loop',
+
+    /**
+     *
+     *
+     */
+    SHOW_WIP_AIRPORTS: 'show-wip-airports'
```
Comment (n8rzz) [inline src/assets/scripts/client/constants/eventNames.js]: remove, not used

**Comment by n8rzz** [comment]
@erikquinn ready for your eyes

File: src/assets/scripts/client/UiController.js [hunk 3]
```diff
@@ -58,6 +65,7 @@ export default class UiController {
         this.$toggleTerrain = null;
         this.$toggleOptions = null;
 
+        prop.ui = ui;
```
Comment (erikquinn) [inline src/assets/scripts/client/UiController.js]: Aren't we trying to kill the `prop`?

File: src/assets/scripts/client/UiController.js [hunk 4]
```diff
@@ -96,13 +105,23 @@ export default class UiController {
...
      * @for UiController
+     * @method setupHandlers
+     * @chainable
+     */
+    setupHandlers() {
+        return this;
+    }
+
+    /**
+     * @for UiController
```
Comment (erikquinn) [inline src/assets/scripts/client/UiController.js]: some descriptions with the docblocks, please.

File: src/assets/scripts/client/UiController.js [hunk 5]
```diff
@@ -293,95 +319,133 @@ export default class UiController {
...
+        this._buildAirportList();
+        this._drawAirportListFooter();
+    }
+
+    /**
+     * @for UiController
+     * @method _buildAirportList
+     * @private
+     */
+    _buildAirportList() {
```
Comment (erikquinn) [inline src/assets/scripts/client/UiController.js]: some descriptions with the docblocks, please.

File: src/assets/scripts/client/UiController.js [hunk 6]
```diff
@@ -293,95 +319,133 @@ export default class UiController {
...
+    }
+
+    /**
+     *  @for UiController
+     * @method _buildAirportListIconForDifficultyLevel
+     * @param difficultyLevel {string}
+     * @return difficulty {string}
+     * @private
+     */
+    _buildAirportListIconForDifficultyLevel(difficultyLevel) {
```
Comment (erikquinn) [inline src/assets/scripts/client/UiController.js]: some descriptions with the docblocks, please. And there's an extra space before the `@for`.

File: src/assets/scripts/client/UiController.js [hunk 7]
```diff
@@ -293,95 +319,133 @@ export default class UiController {
...
+        return difficulty;
     }
 
     /**
      * @for UiController
-     * @method drawAirportListFooter
+     * @method _drawAirportListFooter
      */
-    drawAirportListFooter() {
+    _drawAirportListFooter() {
```
Comment (erikquinn) [inline src/assets/scripts/client/UiController.js]: some descriptions with the docblocks, please.

File: src/assets/scripts/client/UiController.js [hunk 8]
```diff
@@ -293,95 +319,133 @@ export default class UiController {
...
+        const shouldShowWipAirports = this._gameController.getGameOption(GAME_OPTION_NAMES.INCLUDE_WIP_AIRPORTS) === 'yes';
+
+        if (!shouldShowWipAirports) {
+            const notes = $('<span class="words">Additional WIP airports are availalbe by selecting the show all option in the settings window (the gear icon)</span>');
```
Comment (erikquinn) [inline src/assets/scripts/client/UiController.js]: Let's just go for something like `Additional "work-in-progress" airports can be activated in the settings menu`. The long description makes it wrap, which makes the scrollbar appear.

image

File: src/assets/scripts/client/UiController.js (hunk 7 above)
Comment (erikquinn) [inline src/assets/scripts/client/UiController.js]: If practical, you may want to have another look at this method... there seem to be multiple different phrases you want to appear as notes, depending on the state of the option (logical). But they only appear with the ones you want after a page refresh. Opening the airport selection dialog, closing it, changing the setting and reopening it always results in the same message. Only once the page is refreshed do the notes reflect the value of the setting. Not a big deal, but probably not what you wanted either.

File: src/assets/scripts/client/UiController.js (hunk 3 above)
Comment (n8rzz) [inline src/assets/scripts/client/UiController.js]: With a hot hot fire

File: src/assets/scripts/client/UiController.js (hunk 7 above)
Comment (n8rzz) [inline src/assets/scripts/client/UiController.js]: Oops, I pulled out the footer part from the rerendering.

File: src/assets/scripts/client/UiController.js (hunk 4 above)
Comment (n8rzz) [inline src/assets/scripts/client/UiController.js]: done

File: src/assets/scripts/client/UiController.js (hunk 5 above)
Comment (n8rzz) [inline src/assets/scripts/client/UiController.js]: done

File: src/assets/scripts/client/UiController.js (hunk 6 above)
Comment (n8rzz) [inline src/assets/scripts/client/UiController.js]: done

File: src/assets/scripts/client/UiController.js (hunk 7 above)
Comment (n8rzz) [inline src/assets/scripts/client/UiController.js]: done

File: src/assets/scripts/client/UiController.js (hunk 8 above)
Comment (n8rzz) [inline src/assets/scripts/client/UiController.js]: done

**Comment by n8rzz** [comment]
@erikquinn this has been updated and ready for another look when you have time

**Pull request merged**